File: ReglaEspecial.py
from ReglaPrecio import ReglaPrecio


class ReglaEspecial(ReglaPrecio):

    def calcular_total(self, cantidad, precio):
        numero_descuentos = round(cantidad / 3)
        descuento = 0.2 * numero_descuentos
        if(descuento > 0.5):
            descuento = 0.5
        return (precio*cantidad)*(1-descuento)
    
    def es_aplicable(self, sku):
        return sku.startswith("SP")

File: Backend/model/ReglaEspecial.ts
import ReglaPrecio from "./ReglaPrecio";

export default class ReglaEspecial implements ReglaPrecio{

    calcularTotal(cantidad: number, precio: number): number {
        const numeroDescuentos = cantidad/3;
        let descuentoAplicado = 0.2*numeroDescuentos;
        if(descuentoAplicado > 0.5){
            descuentoAplicado = 0.5;
        }
        const precionNormal = cantidad * precio;
        return precionNormal*(1 - descuentoAplicado);
    }

    aplica(sku:string):boolean{
        return sku.startsWith("SP")
    }

}